### PR TITLE
fixed cplus current command

### PR DESCRIPTION
**Pull request by ziklodi**

### CODE REVIEW

File: libexec/cplus-current [hunk 1]
```diff
@@ -24,4 +26,5 @@ esac
 
 commit=$(curl -s $url)
 
-git log -1 --pretty="%C(auto)%h%n%D%Creset" $commit
+git log -1 --pretty="%C(auto)%h%n%D%Creset" $commit 2>/dev/null ||
+  cplus-git-refresh 2>/dev/null && git log -1 --pretty="%C(auto)%h%n%D%Creset" $commit
```
Comment (bru) [inline libexec/cplus-current]: Why not simply calling cplus-git-refresh before the git log command?

**Comment by rviteri** [comment]
I would say just to not run that command if it is not necessary (save some time).

File: libexec/cplus-current (hunk 1 above)
Comment (rviteri) [inline libexec/cplus-current]: I would say just to not run that command if it is not necessary (save some time).

Comment (bru) [inline libexec/cplus-current]: if that's the only option I would still prefer having it as a sequence of commands: it's simpler, less code, yes it has the potential of being slower but:
- if the git repo has been updated recently the time it takes is negligible vs the time it takes to fetch the commit from the remote. in other words in the happy scenario the time it takes is still about the same
- if the git repo does need refreshing, we're actually going to save one command
plus, we are more likely to keep the git repo in sync, which is nice.
plus, we have a much simpler command line :)

Comment (lackac) [inline libexec/cplus-current]: I'm torn between the two options. e.g. I would hate to wait for `cplus current` to return if I want to check production or preprod which don't change that often and their HEAD will probably exist in my local repo at any point. At the same time I see the benefits of having the remote fetched often.

**Pull request merged**